Mathlib/LinearAlgebra/Eigenspace/Matrix.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2024 Jon Bannon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Bannon, Jireh Loreaux
+-/
+
+import Mathlib.LinearAlgebra.Eigenspace.Basic
+
+/-!
+
+# Eigenvalues, Eigenvectors and Spectrum for Matrices
+
+This file collects results about eigenvectors, eigenvalues and spectrum specific to matrices
+over a field.
```

Comment:
I think only the last result needs `R` to be a field. The rest hold under weaker assumptions. Please try to minimize the assumptions you need on `R` by changing them per theorem as needed.

Reply:
Cool. Will do this first thing tomorrow. Thanks!

Reply:
Can you also adjust the docstring accordingly?

Reply:
Certainly! I'll do this now...